crates/ruff_linter/src/rules/ruff/rules/unraw_re_pattern.rs

```diff
@@ -158,8 +162,26 @@ fn check_string(checker: &mut Checker, literal: &StringLiteral, module: RegexMod
     let kind = PatternKind::String;
     let func = func.to_string();
     let range = literal.range;
-    let diagnostic = Diagnostic::new(UnrawRePattern { module, func, kind }, range);
-
+    let mut diagnostic = Diagnostic::new(UnrawRePattern { module, func, kind }, range);
+
+    if
+    // The (no-op) `u` prefix is a syntax error when combined with `r`
+    !literal.flags.prefix().is_unicode()
+        && memchr(
```

Comment:
I don't the we need the performance of memchr here. That's why I would use .contains which is also easier to read

Reply:
It's less about the length of a string and more that the fix code isn't a hot path. 

I don't have too good a sense for when to use memchr otherwise, beyond what the [crate documentation](https://docs.rs/memchr/latest/memchr/#why-use-this-crate) mentions